synthesizer/src/block/transaction/mod.rs

```diff
@@ -84,69 +84,91 @@ impl<N: Network> Transaction<N> {
         vm: &VM<N, C>,
         private_key: &PrivateKey<N>,
         program: &Program<N>,
-        (credits, fee_in_gates): (Record<N, Plaintext<N>>, u64),
+        (credits, fee_in_microcredits): (Record<N, Plaintext<N>>, u64),
         query: Option<Query<N, C::BlockStorage>>,
         rng: &mut R,
     ) -> Result<Self> {
         // Compute the deployment.
         let deployment = vm.deploy(program, rng)?;
         // Compute the fee.
-        let (_, fee, _) = vm.execute_fee(private_key, credits, fee_in_gates, query, rng)?;
+        let (_, fee, _) = vm.execute_fee(private_key, credits, fee_in_microcredits, query, rng)?;
         // Initialize the transaction.
         Self::from_deployment(deployment, fee)
     }
 
-    /// Initializes a new execution transaction from an authorization, and an optional fee.
-    pub fn execute_authorization<C: ConsensusStorage<N>, R: Rng + CryptoRng>(
+    /// Initializes a new execution transaction.
+    pub fn execute<C: ConsensusStorage<N>, R: Rng + CryptoRng>(
         vm: &VM<N, C>,
-        authorization: Authorization<N>,
+        private_key: &PrivateKey<N>,
+        (program_id, function_name): (impl TryInto<ProgramID<N>>, impl TryInto<Identifier<N>>),
```

Comment:
Seems like this maybe was done to emphasize the grouping (and I didn't even know you could group parameter names like this 😹), but I notice it does require you have to supply these as a tuple which is feels a little ergonomically odd and might look off in Rustdoc, maybe we can change back to a list of params?

```suggestion
        program_id, function_name: impl TryInto<ProgramID<N>>
        function_name: impl TryInto<Identifier<N>>),
```

Reply:
The intention is to remove the clippy complaint for too many arguments. Otherwise we need to add a clippy bypass lint onto the function.

Reply:
This convention is commonly used in the `algorithms` crate